app/controllers/coronavirus_local_restrictions_controller.rb

```diff
@@ -43,4 +43,12 @@ def restriction_expiry(search)
 
     [time_until_restriction, MAX_CACHE_TIME].min
   end
+
+  def before_christmas?
+    christmas = Time.zone.parse("2020-12-25")
+
+    !Time.zone.today.after?(christmas)
```

Comment:
This might be simpler to read as `Time.zone.now < Time.zone.parse("2020-12-25")`. I find myself a bit thrown off reading this with `.after` in the condition.

Reply:
I'm never sure if `<` will include Christmas day or not. I'll have a go and see what happens in tests.

Reply:
<= ?

Reply:
Sure, it won't include Christmas. When you do Time.zone.parse without specifying a time it'll default to midnight. So the moment it hits midnight it'll return false.

```
irb(main):005:0> Time.zone.parse('2020-12-24 23:59:59.999') < Time.zone.parse('2020-12-25')
=> true
irb(main):006:0> Time.zone.parse('2020-12-25 00:00') < Time.zone.parse('2020-12-25')
=> false
```

A thing to consider is that for cached responses they'll get this Christmas info for the first 30 mins - but that doesn't seem too big a deal.

Reply:
So, I'm going to change the logic back to use an `unless` and change the method name to `after_christmas?` which I think reads better. As I'm going that, I think it's only to continue using `Time.zone.today.after?(christmas)`

Reply:
Sure thing. Would `after_christmas?` mean >= 00:00 on Boxing day?

Reply:
Yes. Exactly that.

Reply:
Yes!

```
irb(main):001:0> christmas = Time.zone.parse("2020-12-25")
=> Fri, 25 Dec 2020 00:00:00 GMT +00:00

irb(main):003:0> boxing_day = Time.zone.local(2020, 12, 26, 0, 0, 0)
=> Sat, 26 Dec 2020 00:00:00 GMT +00:00
irb(main):004:0> boxing_day.after?(christmas)
=> true

```

Reply:
Sure gotcha - I found myself a bit confused by the Time.zone.today - my brain struggles to realise that you're actually comparing a date object with a time object.